Async hexo generate API endpoint (Sort of addresses #31)

benwiley4000 opened the pull request:
(see https://github.com/redbrick/static-site/issues/31)

API endpoint enables async regeneration of Hexo files (and sending of new emails) without needing to restart the express server.

Bonus - lock file used to prevent multiple concurrent Hexo child processes from spawning. Helps stop nefarious villains from crashing your server!

**New setup requirement:** File called `.env` containing variable `SECRET_API_TOKEN`. Just `cp .env.example .env` and change the token value to anything. If you forget to include `.env` the server won't start.

To trigger hexo generation use:
```
http://localhost:3000/api/regenerate?token=sometoken
```
If it works you'll see: 'Re-generating static site...'
If there's already a hexo process running you'll see: 'Site generation already in progress. Please wait.'
If the token is missing or bad you'll see: 'Bad token.'

You can try it in your browser!

butlerx commented on README.md:
```diff
@@ -15,10 +15,13 @@ Demo on [redbrick.github.io](http://redbrick.github.io), (some links are broken
+
+`mailing_list` and `email_update_log` can be left blank, though updates for every post in history will be sent if no previous send date is specified.
 
 ## Generate
```
could you also add to the reame the process for regenerateing using the api endpoint

butlerx commented on app.js:
```diff
@@ -102,6 +105,52 @@ app.get(baseUrl + 'fetchMeSomeTea', function (req, res) {
+        ['generate']
+      );
+      hexoGenerate.stdout.on('data', function (buffer) {
+        console.log(buffer.toString())
```
missing semicolon

benwiley4000 commented:
I updated the readme, fixed the semicolon, and also decided to move the `hexo_lock` deletion until after emails are sent, since concurrent email processes could cause duplicate emails to be sent to the mailing list.
